lib/aruco_detector.hpp

```diff
@@ -0,0 +1,55 @@
+#include <opencv2/opencv.hpp>
+#include <opencv2/aruco.hpp>
+#include <memory>
+#include "classification_model.hpp"
+
+class Aruco_Detector : public ClassificationModel::dClassification {
+public:
+    Aruco_Detector(cv::VideoCapture& videoCapture, int dictionaryType) : inputVideo(videoCapture), dictionaryType(dictionaryType) {}
+
+    ClassificationModel::Detection classify(cv::Mat& image, float threshold) override {
```

Comment:
The `classifiy` and `get_bounding_box` implementations do not conform to the ClassificationModel type-erased ones. Look at yolov8_arrow.hpp for an example.